agent/mission_solver/data_loader.py

```
@@ -0,0 +1,158 @@
+from .models import QuestPhase, Mission, MissionCond
+
+logger = logging.getLogger("MissionSolver")
+
+# 数据目录: 与当前模块同级目录
+RESOURCE_DIR = os.path.dirname(os.path.abspath(__file__))
+
+
+def _load_json(filename: str) -> dict | list:
+    filepath = os.path.join(RESOURCE_DIR, filename)
+    if not os.path.exists(filepath):
+        raise FileNotFoundError(f"数据文件不存在: {filepath}\n请先运行: python tools/update_quest_data.py")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)